src/wallet/wallet.ts

```diff
@@ -110,24 +132,57 @@ export class Wallet implements IWallet {
         readonly boardingTapscript: DefaultVtxo.Script,
         readonly serverUnrollScript: CSVMultisigTapscript.Type,
         readonly forfeitOutputScript: Bytes,
-        readonly dustAmount: bigint
-    ) {}
+        readonly dustAmount: bigint,
+        walletRepository: WalletRepository,
+        contractRepository: ContractRepository
+    ) {
+        this.walletRepository = walletRepository;
+        this.contractRepository = contractRepository;
+    }
```

Comment:
```suggestion
private constructor(
readonly identity: Identity,
readonly network: Network,
@@ -110,24 +132,57 @@
readonly boardingTapscript: DefaultVtxo.Script,
readonly serverUnrollScript: CSVMultisigTapscript.Type,
readonly forfeitOutputScript: Bytes,
        readonly dustAmount: bigint,
        readonly walletRepository: WalletRepository,
        readonly contractRepository: ContractRepository
    ) {}
```

public fields can be delcared in constructor